distribution/build.gradle

```diff
@@ -120,70 +110,181 @@ configure(subprojects.findAll { it.name == 'zip' || it.name == 'tar' }) {
       with copySpec {
         with commonFiles
         from('../src/main/resources') {
-          include 'bin/*.bat' 
+          include 'bin/*.bat'
           filter(FixCrLfFilter, eol: FixCrLfFilter.CrLf.newInstance('crlf'))
         }
         MavenFilteringHack.filter(it, expansions)
       }
       from('../src/main/resources') {
-        include 'bin/*.exe' 
-      } 
+        include 'bin/*.exe'
+      }
     }
   }
 }
 
 /*****************************************************************************
  *                         Deb and rpm configuration                         *
- *****************************************************************************/
-// ospackage supports adding empty dirs with directory() to rpm, but not deb...yet
-// https://github.com/nebula-plugins/gradle-ospackage-plugin/issues/115
-// however, even adding just for rpm doesn't seem to work...
-// gradle may also get native support https://issues.gradle.org/browse/GRADLE-1671
-// in the meantime, we hack this by copying an empty dir
-// TODO: HACK DOES NOT WORK
-/*ext.emptyDir = new File(project.buildDir, 'empty') 
-Closure emptyDirSpec() {
-  return {
-    from emptyDir
-    addParentDirs false
-    createDirectoryEntry true
+ *****************************************************************************
+ *
+ * The general strategy here is to build a directory on disk, packagingFiles
+ * that contains stuff that needs to be copied into the distributions. This is
+ * important for two reasons:
+ * 1. ospackage wants to copy the directory permissions that it sees off of the
+ *    some filesystem. If you ask it to create a directory that doesn't already
+ *    exist on disk it petulantly creates it with 0755 permissions, no matter
+ *    how hard you try to convince it otherwise.
+ * 2. Convincing ospackage to pick up an empty directory as part of a set of
+ *    directories on disk is reasonably easy. Convincing it to just create an
+ *    empty directory requires more wits than I have.
+ * 3. ospackage really wants to suck up some of the debian control scripts
+ *    directly from the filesystem. It doesn't want to process them through
+ *    MavenFilteringHack or any other copy-style action.
+ */
+configure(subprojects.findAll { it.name == 'deb' || it.name == 'rpm' }) {
+  def packagingFiles = new File(project.buildDir, 'packaging')
+  project.ext.packagingFiles = packagingFiles
+  task processPackagingFiles(type: Copy) {
+    from '../src/main/packaging'
+    from 'src/main/packaging'
+
+    MavenFilteringHack.filter(it, expansions)
+    into packagingFiles
+    /* Explicitly declare the outputs so that gradle won't skip this task if
```

Comment:
It feels more right to declare the stuff is making rather than explicitly setup a chain.